src/Orchard.DisplayManagement/Descriptors/DefaultShapeTableManager.cs

```diff
@@ -133,7 +133,7 @@ public ShapeTable GetShapeTable(string themeName)
             return builtAlterations;
         }
 
-        private async void BuildDescriptorsAsync(IShapeTableProvider bindingStrategy, IEnumerable<ShapeAlteration> builtAlterations)
+        private async Task<Task> BuildDescriptorsAsync(IShapeTableProvider bindingStrategy, IEnumerable<ShapeAlteration> builtAlterations)
         {
             var alterationSets = builtAlterations.GroupBy(a => a.Feature.Descriptor.Id + a.ShapeType);
```

Comment:
I see a Task.Run in this code, I am not really confident with it, how many threads could it start in the worst case? Also aren't alterations supposed to run in order, and multi-threading would just break this?